external/Makefile

```diff
@@ -45,18 +45,19 @@ CSRCS =
 VPATH = .
 
 
-ifeq ($(CONFIG_DM),y)
-ifeq ($(CONFIG_LWM2M_WAKAAMA),y)
+ifneq ($(CONFIG_DM)$(CONFIG_EXTERNAL_WAKAAMA_EXAMPLE),)
 include wakaama/core/Make.defs
```

Comment:
Also, with the dependency on CONFIG_DM and CONFIG_LWM2M_WAKAAMA implied, the check for CONFIG_DM appears to be redundant here. It is perhaps enough to check only for WAKAAMA, which makes sense as you are trying to build from wakaama source files here.